src/compiler/crystal/codegen/link.cr

```diff
@@ -104,6 +104,53 @@ module Crystal
       ENV.fetch("CRYSTAL_LIBRARY_RPATH", "")
     end
 
+    # Adds the compiler itself's RPATH to the environment for the duration of
+    # the block. `$ORIGIN` in the compiler's RPATH is expanded immediately, but
+    # `$ORIGIN`s in the existing environment variable are not expanded. For
+    # example, on Linux:
+    #
+    # - CRYSTAL_LIBRARY_RPATH of the compiler: `$ORIGIN/so`
+    # - Current $CRYSTAL_LIBRARY_RPATH: `/home/foo:$ORIGIN/so`
```

Comment:
It would be easier to follow the example if `$ORIGIN/so` wasn't used in both sources. Same for dll.